windows_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2010 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build windows
+// +build windows
+
+package fsnotify
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestRemovalOfWatchTwice(t *testing.T) {
+	// Create directory to watch
+	testDir := tempMkdir(t)
+	defer os.RemoveAll(testDir)
+
+	// Create a file before watching directory
+	testFileAlreadyExists := filepath.Join(testDir, "TestFsnotifyEventsExisting.testfile")
+	{
+		var f *os.File
+		f, err := os.OpenFile(testFileAlreadyExists, os.O_WRONLY|os.O_CREATE, 0666)
+		if err != nil {
+			t.Fatalf("creating test file failed: %s", err)
+		}
+		f.Sync()
+		f.Close()
+	}
+
+	watcher := newWatcher(t)
+	defer watcher.Close()
+
+	addWatch(t, watcher, testDir)
+	if err := watcher.Remove(testDir); err != nil {
+		t.Fatalf("Could not remove the watch: %v\n", err)
+	}
+	if err := watcher.remWatch(testDir); err == nil {
```

Comment:
If this line was also` watcher.Remove` I think this should be safe to put back in as an integration test across all OSes. It works correctly on my as-yet-not-merged illumos tree, and I think we should do our best that most tests should be run across all platforms rather than specific to a single one...

Reply:
Okay, added

Reply:
Uhhh. Those test failures are confusing me. I wonder if the official API is to allow repeated removal and just ignore it if it's not watched? Let me do some digging. Thank you for being responsive! Sorry for the delay, but I think we can get this right.